src/monitor.rs

```diff
@@ -33,33 +33,49 @@ impl AppState {
 
     /// Poll the Ethereum node for the latest block and update the shared state
     /// with the latest block.
-    #[tracing::instrument(skip(app_state))]
+    #[tracing::instrument(skip_all)]
     pub(crate) async fn poll_and_update_block<
         T: Transport + Clone,
         P: Provider<T, N> + std::fmt::Debug,
         N: Network,
     >(
+        block: Block,
         provider: P,
```

Comment:
I think we should be able to pass the provider as a reference here since we are not spawning any tasks (we just await the call in the current task).